lib/configurators/aws/ec2discovery.go

```diff
@@ -14,10 +14,25 @@
 
 package aws
 
-import "fmt"
+import (
+	crand "crypto/rand"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+	mrand "math/rand"
+	"time"
+)
 
 func EC2DiscoverySSMDocument(proxy string) string {
-	return fmt.Sprintf(ec2DiscoverySSMDocument, proxy)
+	randomBytes := make([]byte, 4)
+	if _, err := crand.Read(randomBytes); err != nil {
+		// on error from crypto rand fallback to less secure math random
+		mathRand := mrand.New(mrand.NewSource(time.Now().UnixNano()))
+		binary.LittleEndian.PutUint32(randomBytes, mathRand.Uint32())
+	}
+	randString := hex.EncodeToString(randomBytes)
+
+	return fmt.Sprintf(ec2DiscoverySSMDocument, randString, proxy, randString)
```

Comment:
I considered trying to template this more, however there is additional templating in the document using `{{`, which we would need to not interact with.  So I felt like for now this is probably the best option.

Reply:
+1 for using a template. It's hard to tell when looking at code like this whether or not the Sprintf is valid, because in order to get the full picture you have to:

- navigate to the definition of `ec2DiscoverySSMDocument` 
- keep track of how many format verbs there are and in what order they appear
- navigate back to the Sprintf and make sure it has the right number of args and that they are provided in the right order

Reply:
My 2c: I don't mind as much, in this case, now that the format is right there in the Sprintf call. Even string concatenation could work here, as it's relatively simple.